### PR TITLE
hypre: add cublas and rocblas variants

### PR DESCRIPTION
**Pull request by jrood-nrel**
We've had these options in a custom hypre package for a while. Moving them here to see if anyone would like them in Spack.

I think this also fixes a mistake with GPU aware MPI.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hypre/package.py
Comment (alecbcs) [inline var/spack/repos/builtin/packages/hypre/package.py]:
```suggestion
            if spec.satisfies("+cublas"):
                conigure_args.append("--enable-cublas")
```

It's now recommended to use `spec.satisfies()` instead of `in`.

File: var/spack/repos/builtin/packages/hypre/package.py
Comment (alecbcs) [inline var/spack/repos/builtin/packages/hypre/package.py]: It's now recommended to use `spec.satisfies()` instead of `in`.

**Review by alecbcs — CHANGES_REQUESTED** [review]
PR looks great! Just a few suggestions to follow the recommended use of `satisfies()`.

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks @jrood-nrel for the super quick follow up to feedback.

I'll defer to @psakievich or a package maintainer for a final review before merging.

**Review by psakievich — APPROVED** [review]
LGTM. ~~@alecbcs will you mark your comments as resolved? I believe we have to do that now for merging.~~ I'm satisfied but ideally I would also like to see a package maintainer approve this.

**Review by osborn9 — APPROVED** [review]

**Pull request merged**
